### DISCUSSION TITLE
Quarkus Command line Application

### DISCUSSION THREAD

**Discussion by omprakashyadav** [quarkusio/quarkus#24116] [Q&A] [answered]
Hello Everyone,

I am developing an Quarkus CLI application, so that I can run it on the AWS Batch or AWS Lambda inside a container todo some processing.

In my application, I am using command pattern.
```java
class DoHeavyWork {

 @Inject Logger logger;
 @S3Client s3Client

public  DoHeavyWork(String s3FilePath){
  this.s3FilePath=s3FilePath;
}

public void execute(){
   logger.info() -> Throws NULL Pointer Exception.
}
}


```
In my build.gradle, I am using: Picoli extension.
`implementation 'io.quarkus:quarkus-picocli'`

Now, when I am launching the command like:
`new DoHeavyWork(args).execute();`

This is throwing an NPE saying logger is null.

Bean injection is not working because, I am creating a new class with `new`  keyword.
I there a way to inject the logger and other beans at runtime after creating a object with new keyword.
I want to use new keyword instead of injecting the Command.

Is there a way to do this????

**Comment by maxandersen** [comment]
No. By using new you are not making it possible for any framework to do injection.

**Reply by omprakashyadav** [reply]
Is there any way to perform the dynamic injection?

Like using CDI.current() or BeanManager. 

Thanks 
Om

**Comment by omprakashyadav** [comment, marked as answer]
I have found a way to get the CDI classes by using the following code:
`CDI.current().select(BlobReaderFactory.class).get();`
